chia/full_node/full_node_api.py

```diff
@@ -1463,13 +1457,8 @@ async def register_interest_in_puzzle_hash(
             hint_coin_ids.extend(ph_hint_coins)
             if puzzle_hash not in self.full_node.ph_subscriptions:
                 self.full_node.ph_subscriptions[puzzle_hash] = set()
-            if (
-                peer.peer_node_id not in self.full_node.ph_subscriptions[puzzle_hash]
-                and self.full_node.peer_sub_counter[peer.peer_node_id] < max_items
-            ):
+            if self.full_node.peer_subscriptions_count(peer) < max_items:
```

Comment:
`peer_subscription_count()` is O(n) in your patch, and it's called in a loop computing almost the same value over and over again. you could hoist this out of the loop, but the wallet may still be making these requests in a loop, and you're basically back to what you have now.

it would seem reasonable to keep the counter in the peer object itself. We don't preserve the subscriptions across disconnects, right?